src/main/java/org/springframework/integration/aws/inbound/kinesis/KclMessageDrivenChannelAdapter.java

```diff
@@ -189,7 +201,7 @@ protected void onInit() {
 						this.kinesisProxyCredentialsProvider,
 						null, null,
 						KinesisClientLibConfiguration.DEFAULT_FAILOVER_TIME_MILLIS,
-						UUID.randomUUID().toString(),
+						(this.workerId == null) ? UUID.randomUUID().toString() : this.workerId,
```

Comment:
That's not correct.

Please, make `workerId` property with `UUID` by default and add `Assert.hasText()` into the `setWorkerId()`.
This way we protect ourselves from the `null` and won't have such a code here in the `onInit()`.